codalab/worker/bundle_manager.py

```diff
@@ -529,3 +534,24 @@ def _construct_run_message(self, worker, bundle):
         resources['request_network'] = bundle.metadata.request_network
 
         return message
+
+    def _fail_unresponsive_bundles(self):
+        """
+        Fail bundles in uploading, staged and running state if we haven't heard from them for more than
+        BUNDLE_TIMEOUT_DAYS days.
+        """
+        failure_message = "Bundle hasn't respond in {} days.".format(BUNDLE_TIMEOUT_DAYS)
```

Comment:
Make message more specific: Bundle has been stuck in {state} for {} days, failing.